File: daily/21.11.23/1107.py
import sys
input = sys.stdin.readline

n = int(input())
m = int(input())
broken = set(input().strip())

now = 100
num = 10**8

if now == n:
    print(0)
elif m == 10:
    print(abs(n - now))
elif m == 0:
    print(min(len(str(n)), abs(n - now)))
else:
    for i in range(1000000):
        for k in str(i):
            if k in broken:
                break
        else:
            if i == n:
                num = n
                break
            if abs(num - n) > abs(i - n):
                num = i
            else:
                break

    print(min( (len(str(num)) + abs(num - n)), abs(n - now) ))

File: solved.ac/CLASS3_JS/1107.js
const fs = require("fs");
const inputPath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const stdin = fs
  .readFileSync(inputPath)
  .toString()
  .split("\n")
  .map((line) => line.trim());

const input = (() => {
  let line = 0;
  return () => stdin[line++];
})();

const check = (num, ban) => {
  const strNum = num.toString();

  for (let i = 0; i < ban.length; i++) {
    if (strNum.includes(ban[i])) {
      return false;
    }
  }
  return true;
};

const solution = () => {
  const n = input();
  const m = Number(input());
  let ans = Math.abs(Number(n) - 100);
  let arr = [];
  if (m) {
    arr = input().split(" ").map(Number);
  }

  for (let i = 0; i <= 1000000; i++) {
    if (check(i, arr)) {
      ans = Math.min(ans, i.toString().length + Math.abs(Number(n) - i));
    }
  }
  console.log(ans);
};

solution();
